athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBMetadataHandler.java

```diff
@@ -217,9 +218,51 @@ public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesReque
         if (DynamoDBConstants.DEFAULT_SCHEMA.equals(request.getSchemaName())) {
             combinedTables.addAll(tableResolver.listTables());
         }
-        return new ListTablesResponse(request.getCatalogName(), new ArrayList<>(combinedTables), null);
+        return new ListTablesResponse(request.getCatalogName(), new ArrayList<>(filterTables(combinedTables)), null);
     }
 
+ /**
+     * @author vaquar khan
+     * This method is used to filter DyanamoDB tables in Athena 
+     * We enable users to configure optional environment variables with commaseparated List
+     * dynamo_connector_filter (Optional) Can be used to fetch selected tables from DynamoDB.
+     * default the connector will attempt to get all DynamoDB tables while connecting via Athena.(e.g. *  or tbl1,tbl2,tbl3)
+     * @param combinedTables
+     * @return
+     */
+    
+	private Set<TableName> filterTables(Set<TableName> combinedTables)
+	{
+		String dynamoDBFilter = "*";
+		Set<TableName> filterCombinedSetTables = new LinkedHashSet<>();
+		String database = DynamoDBConstants.DEFAULT_SCHEMA;
+		List<String> list = null;
+		boolean flag = false;
+
+		dynamoDBFilter = System.getenv("dynamo_connector_filter");
+
+		if (null == combinedTables || null == dynamoDBFilter) {
+			return combinedTables;
+		}
+
+		if (null != dynamoDBFilter) {
+			flag = dynamoDBFilter.toUpperCase().contains("*");
```

Comment:
What if the filter is set to: `sometable1,sometable2*,sometable3` ?

I think either we should throw an exception here or implement actual wildcard or regex matching if you want to support the case where a `*` is contained in the filter rather than equals the filter.